File: 0389-find-the-difference/0389-find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        c = Counter(t)

        for char in s:
            c[char] -= 1
            if c[char] == 0:
                c.pop(char)

        return c.popitem()[0]  

File: 0389-find-the-difference/0389-find-the-difference.cpp
const auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    const char findTheDifference(const string s, const string t) {
        unordered_map<char, int> sMap;
        for (const char& c : s) ++sMap[c];
        for (const char& c : t) {
            if (!sMap.contains(c)) return c;
            --sMap[c];
            if (sMap[c] == 0) sMap.erase(c);
        }
        return 'a';
    }
};